src/Hazelcast.Net/Serialization/Compact/ReflectionSerializer.cs

```diff
@@ -528,5 +534,26 @@ public virtual void Write(ICompactWriter writer, object obj)
             for (var i = 0; i < x.Length; i++) r[i] = x.GetValue(i)?.ToString();
             return r;
         }
+
+        private static void VerifyTypeIsSupported(object o)
+        {
+            // for FullName to be null, the type would need to be derived from an open generic somehow,
+            // which makes no sense since it is the actual type of a concrete object. we can assume that
+            // type is not going to be null.
+            var type = o.GetType();
+            var name = type.FullName!;
```

Comment:
I can't image even an open type name could be null 🤔

Reply:
Point is, it *is* possible to override `GetType()` and return a class that derives from `Type` but provides a `null` `FullName`. I have no idea why this is possible nor why would anyone do it, but nevertheless it *is* possible and hence nullable checks raise a warning. I choose to ignore it with the `!`.